controllers/nifidataflow_controller.go

```diff
@@ -100,6 +104,7 @@ func (r *NifiDataflowReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	// Check if the cluster reference changed.
 	original := &v1alpha1.NifiDataflow{}
 	current := instance.DeepCopy()
+	patchCurrent := client.MergeFrom(current.DeepCopy())
```

Comment:
In fact `current` and `instance` are different because we update the `ClusterRef` on instance: https://github.com/konpyutaika/nifikop/blob/master/controllers/nifidataflow_controller.go#L105